plutus-contract/src/Plutus/Trace/Emulator.hs

```diff
@@ -249,6 +269,17 @@ runEmulatorTrace cfg trace =
     $ foldEmulatorStreamM (generalize list)
     $ runEmulatorStream cfg trace
 
+runEmulatorTrace'
```

Comment:
I think you want to refactor `runEmulatorTrace` following `runEmulatorTrace'`. Or just replace one with the other.

Reply:
Thanks, which the old version I couldn't do that, because of the uncertainty of the initial thread returning or not. With the `exit` mechanism the uncertainty is not there so everything can use the same function.